rtgui/thumbnail.cc

```diff
@@ -205,53 +204,40 @@ void Thumbnail::_generateThumbnailImage ()
     cfs.exifValid = false;
     cfs.timeValid = false;
 
-    if (ext == "jpg" || ext == "jpeg") {
-        infoFromImage (fname);
-        tpp = rtengine::Thumbnail::loadFromImage (fname, tw, th, -1, pparams->wb.equal, pparams->wb.observer);
-
-        if (tpp) {
-            cfs.format = FT_Jpeg;
-        }
-    } else if (ext == "png") {
-        tpp = rtengine::Thumbnail::loadFromImage (fname, tw, th, -1, pparams->wb.equal, pparams->wb.observer);
+    // RAW works like this:
+    //  1. if we are here it's because we aren't in the cache so load the JPG
+    //     image out of the RAW. Mark as "quick".
+    //  2. if we don't find that then just grab the real image.
+    bool quick = false;
 
-        if (tpp) {
-            cfs.format = FT_Png;
-        }
-    } else if (ext == "tif" || ext == "tiff") {
-        infoFromImage (fname);
-        tpp = rtengine::Thumbnail::loadFromImage (fname, tw, th, -1, pparams->wb.equal, pparams->wb.observer);
+    rtengine::eSensorType sensorType = rtengine::ST_NONE;
+    if ( initial_ && options.internalThumbIfUntouched) {
+        quick = true;
+        tpp = rtengine::Thumbnail::loadQuickFromRaw (fname, sensorType, tw, th, 1, TRUE);
+    }
 
-        if (tpp) {
-            cfs.format = FT_Tiff;
-        }
-    } else {
-        // RAW works like this:
-        //  1. if we are here it's because we aren't in the cache so load the JPG
-        //     image out of the RAW. Mark as "quick".
-        //  2. if we don't find that then just grab the real image.
-        bool quick = false;
-
-        rtengine::eSensorType sensorType = rtengine::ST_NONE;
-        if ( initial_ && options.internalThumbIfUntouched) {
-            quick = true;
-            tpp = rtengine::Thumbnail::loadQuickFromRaw (fname, sensorType, tw, th, 1, TRUE);
-        }
+    if ( tpp == nullptr ) {
```

Comment:
`if (!tpp) {`

Reply:
Okay... but... I didn't write that line...
```
$ git blame -L241,+1 0fdac3797aebbba4a751184e60278e5460bc56b7 -- rtgui/thumbnail.cc
0731975ff0 (Flössie 2016-10-12 17:48:40 +0200 241)         if ( tpp == nullptr ) {
```

Reply:
I know, I know. But while you're at it...

And actually, I didn't write that line either, just fired clang-tidy at it. :smiley:

Reply:
There are five instances of `tpp == nullptr` and two of `!tpp`.  Should I change them all?